Add Return Value Support for Script Functions

ShinyWasabi opened the pull request:
- Added return value support for script functions and exposed it to Lua.
- Removed Dance Mode & Los Santos Customs.
- Replaced add_clan_logo_to_vehicle with a script function.
- Updated creator stack size for start_creator_script.
- Fixed reset_session_data not being called due to pausemenu_multiplayer being terminated immediately after the we request it. The function is now called from main_persistent.
- Updated load_from_datafile pattern.
- Added get_int & set_int methods to the Lua memory table.

Closes https://github.com/YimMenu/YimMenu/issues/3417
Closes #3497

ShinyWasabi committed:
Add Return Value Support for Script Functions

Commit:
Merge branch 'master' into refactor-sc-functions

maybegreat48 commented on src/backend/backend.cpp:
```diff
@@ -54,7 +54,6 @@ namespace big
 		{
 			looped::self_wanted();
 			looped::self_hud();
-			looped::self_dance_mode();
```
Why was this removed?

Mr-X-GTA replied:
It works for me, but the `dance_state` global seems to be incorrect (should be `1943513`?)

ShinyWasabi replied:
The main reason it was removed isn’t solely because it wasn’t working, but because these features can now be converted into a Lua script, as we have exposed script patches and functions. The same applies to LSC. They don't need to be in the base YimMenu.

sch-lda replied:
This feature is used very infrequently and convert it to Lua should have little impact. Please try not to convert frequently used features that are already located within YimMenu to Lua, not all users follow the PR and Lua repositories and they don't know where to find these features. These features are already localized in other languages, and converting them to Lua will make multilingual support more difficult.

Tayipc38 replied:
Can't we add a little window to download these Lua's via the menu itself?

Yimura replied:
I agree with both, however some functionality is quite shit and should be considered for removal whereas some of the others can be reworked a bit and kept in.

Also at some point we should look into making a sort in-menu Lua workshop to install scripts with.

Tayipc38 replied:
> Also at some point we should look into making a sort in-menu Lua workshop to install scripts with.

I can do that right now

Commit:
Fixed lua/scr_function.

Commit:
Merge branch 'master' into refactor-sc-functions

Commit:
Removed address-of operator on instruction_pointer parameter, as sol is pass-by-value.

ShinyWasabi committed:
Fixed reset_session_data & start_creator_script

ShinyWasabi committed:
Added support for Vector3 and updated casting for bool type for proper conversion to Lua boolean

ShinyWasabi committed:
Updated documentation for scr_function

Commit:
Fixed improper casting of Boolean return type.

Commit:
Updated function_name documentation.

ShinyWasabi committed:
Added get_int method and updated param names for script functions

Commit:
Fix #3497 graceful landing not saved.

Commit:
Added a check in view_lsc to see if the vehicle can accept clan logos first.

ShinyWasabi committed:
Added script patch for the SP check of vehicle_cannot_accept_clan_logo

Commit:
Fixed vehicle clan logo SP bypass not working properly.

Commit:
Fixed COPY VEHICLE not giving persist_car_service::spawn_vehicle_json the target's ped so it can copy their clan logo and not ours.
Fixed spawn_vehicle_json calling add_clan_logo_to_vehicle with our logo and not the ped parameter's logo.

xiaoxiao921 commented on src/script_function.hpp:
```diff
@@ -1,43 +1,71 @@
+		}
+	};
+
+	static inline script_function set_as_ceo("SAC", "freemode"_J, "2D 04 1D 00 00 5D");
```
static inline for global variables in hpp file mean that for every include the ctor is fired. just make them inline.

Commit:
Code review changes.

maybegreat48 commented on src/util/scr_functions.hpp:
```diff
@@ -0,0 +1,41 @@
+
+#include "script_function.hpp"
+
+namespace big::scr_functions
```
Can't we just call the script functions directly? And I think we should store them in `scr_functions` instead of leaving it at `big`

Commit:
Code review changes.

Commit:
Merge branch 'master' into refactor-sc-functions

Commit:
Added Clone Player Car.

Commit:
Fixed has_clan_logo check in view_lsc being given the wrong parameter.
